### DISCUSSION TITLE
Add option to continue through all candidates and pick highest similarity, rather than stopping at first threshold breach

### DISCUSSION THREAD

**Discussion by laker-93** [beetbox/beets#5036] [Q&A]
I have disabled missing tracks negatively impacting a match in beets as often I do not want to import the entire album, or do so incrementally. This has been working well however I have noticed that beets will sometimes mis match a track import as if a track's title matches an earlier track in the album to a degress that satisfies the import threshold, it will match to the earlier track instead of continuing through and finding the right match later in the album.
For example, see here when importing track 11. This gets mis matched to track 2. If beets carried on through the album until track 11 it would have found a perfect match.
I do not want to raise the match threshold to 100% as I need it lower for other uses. I want to be able to specify an option to force beets to continue through an album match track by track and pick the highest match overall, rather than stopping at the first.

```
/downloads
/music/Skee Mask/B/11 - Portal Perc.mp3 (3 items)
Tagging:
    Skee Mask - B
URL:
    https://musicbrainz.org/release/2c0e7e2b-a4ad-4b37-b698-3d76534ebc9f
(Similarity: 91.8%) (tracks) (Digital Media, 2022, XW)
 * Portal Perc (#11) -> Steamer (Early mix) (#2) (title, index)
Missing tracks (8/11 - 72.7%):
 ! P. Prog          (# 1) (5:37)
 ! Steamer (AD mix) (# 3) (5:49)
 ! mWidit (Edit)    (# 4) (5:39)
 ! MFB606Delay      (# 5) (4:54)
 ! AD Dub           (# 6) (6:28)
 ! 448              (# 7) (4:54)
 ! Jambient         (# 8) (5:49)
 ! JD 372           (# 9) (5:47)
```

**Comment by sampsyo** [comment]
Hmm; this is actually not an explicit beets policy (to prefer earlier matches over later ones, in the track ordering). This is probably happening because of other factors, not because of any intentional "earlier is better" preference in the matching.

One thing I'm confused about in this specific case: it looks like track 11 is not missing? That is, it seems like some _other_ file was matched as track 11. That could explain why this one was remapped to something else…
